InterviewQuestions/Chapters/ArraysStrings/ArraysAndStringSolutions.cs

```diff
@@ -88,5 +88,45 @@ public bool isStringPermutation(string firstString,string secondString)
             return flag;    
 
         }
+        public int[,] zeroesMatrix(int[,] matrix)
+        {
+            int row=matrix.GetLength(0);
+            int col=matrix.GetLength(1);
+            int[,] testMatrix= new int[row,col];
+            for (int i = 0; i < row; i++)
+            {
+                for (int j = 0; j < col; j++)
+                {
+                    matrix[i, j] = testMatrix[i, j];
+                }
+            }
+            for (int i=0; i < row; i++)
+            {
+                for(int j=0; j<col; j++)
+                {
+                    if (matrix[i,j] == 0)
+                    {
+                        for(int k=0; k<row; k++)
+                        {
+                            testMatrix[i,k] = 0;
+                        }
+                    }
+                }
+            }
+            for (int i = 0; i < row; i++)
+            {
+                for (int j = 0; j <col; j++)
+                {
+                    if (matrix[i,j] == 0)
+                    {
+                        for (int k = 0; k < col; k++)
+                        {
+                            testMatrix[k,j] = 0;
+                        }
+                    }
+                }
+            }
+            return testMatrix;
+        }

```

Comment:
There are too many loops in the solution. Be sure that if you already convert elements to 0 for the specific column, when you check the next row, the element that belongs to the specific column will appear as 0.
In this case, you should not try to convert to 0 all column elements which are already converted in the previous step.

Hint: Loop through all arrays once and hold the indexes of the elements which are 0. And then try to convert this specific column and row to zero.

Reply:
three nested loops are not allowed. Please modify your code to have max two nested loops.

#Hint: Loop through whole matrix ones and hold the indexes of the '0's and then nullify these rows and columns only, do not loop through the whole matrix again.

Reply:
I changed the solution way for less loop. Look here 9ee014712e743b2ad08497d656507c739ffdc6b5

Reply:
nice fix!